core/src/main/java/io/questdb/griffin/engine/functions/bind/BindVariableServiceImpl.java

```diff
@@ -969,28 +969,28 @@ private static void setStr0(Function function, CharSequence value, int index, @N
                     ((CharBindVariable) function).value = (char) (Chars.nonEmpty(value) ? Numbers.parseShort(value) : 0);
                     break;
                 case ColumnType.INT:
-                    ((IntBindVariable) function).value = value != null ? Numbers.parseInt(value) : Numbers.INT_NaN;
+                    ((IntBindVariable) function).value = Chars.nonEmpty(value) ? Numbers.parseInt(value) : Numbers.INT_NaN;
```

Comment:
With this change we treat `stmt.setInt(1, "")` as `stmt.setNull(1, Types.INTEGER)`. Previously the first call would result into a `NumericException` and, thus, an error returned back to the client. So, I wonder if we need this breaking change?

The same consideration applies to the rest of this method.

Reply:
> we treat `stmt.setInt(1, "")`

Sorry for the confusing example - probably I didn't wake up fully when I was writing this 🙂. Obvisouly, this code won't compile. I still wonder when we receive an empty string instead of a null. Could you clarify?

Reply:
the only way of hitting this method is with code like `ps.setString(index, String)`. For the case of the tests

```
sink.clear();
                try (PreparedStatement ps = connection.prepareStatement("tab1 where ? is not null")) {
                    ps.setString(1, "");
                    try (ResultSet rs = ps.executeQuery()) {
                        assertResultSet(
                                "value[INTEGER],ts[TIMESTAMP]\n",
                                sink,
                                rs
                        );
                    }
                }
```
with this change we treat both null and and empty string as null, which after closer inspection (`StrConstant`) seems like a compatibility break, will rebert

Reply:
It seems to be related with what @bziobrowski described in #1980: the above snippet leads to the default type (DOUBLE) being used instead of the one passed by the driver (STRING). We should deal with this problem separately.

Reply:
indeed, in this case the function parser is not seeing `is` it is seeing `=` with args `IndexedParameterLinkFunction` (type string) and `NullConstant`. It is not making use of the hint apparently. I would rather address this in a follow up PR. This one deals with IS [NOT] NULL, small enough scope.

Reply:
Agreed - it's all result of not handling prepared statement bind variable types properly. 

Assuming 
create table cc as (x integer); 
in PG it looks like this 
`
PreparedStatement stmt = connection.prepareStatement("select x from cc where x = ? "));
stmt.setString(1, ""); //or stmt.setString(1,null)
stmt.executeQuery();
`

both produce : 
_org.postgresql.util.PSQLException:** ERROR: operator does not exist: integer = character varying
                    //  Hint: No operator matches the given name and argument types. You might need to add explicit type_ 

`
select * from tab1 where ? is not null 
`
works fine in PG because pstmt bind variables types are used and PG knows it varchar or some other type . 
This all means that this change is basically fixing scenario that shouldn't occur if #1980 is implemented.